### PR TITLE
fix(rust): decompress the right number of rows when reading compressed CSVs

### PR DESCRIPTION
**Pull request by Wainberg**
Closes https://github.com/pola-rs/polars/issues/10584.

### CODE REVIEW

**Comment by alexander-beedie** [comment]
Some lint to deal with; also can you add some unit tests (in `py-polars/tests/unit/io/test_csv.py`) to help demonstrate the issue and confirm the fix 😉👍

**Comment by Wainberg** [comment]
@alexander-beedie fixed the issues and added a test, so hopefully we should be good to merge :)

File: py-polars/tests/unit/io/test_csv.py [hunk 1]
```diff
@@ -1737,3 +1737,17 @@ def test_invalid_csv_raise() -> None:
     "SK0127960V000","SK BT 0018977","
     """.strip()
         )
+
+
+@pytest.mark.slow()
+def test_partial_read_compressed_web_file() -> None:
+    url = "https://raw.githubusercontent.com/hadley/mastering-shiny/main/neiss/injuries.tsv.gz"
```
Comment (stinodego) [inline py-polars/tests/unit/io/test_csv.py:1744]: The test suite should not be reliant on external content if at all possible. Please include a small example (e.g. a multiline string) and read from that.

**Review by stinodego — CHANGES_REQUESTED** [review]
I didn't look at the issue/implementation, but I have a comment about the test.

File: py-polars/tests/unit/io/test_csv.py (hunk 1 above)
Comment (Wainberg) [reply py-polars/tests/unit/io/test_csv.py:1744]: This issue only triggers when reading a compressed file with a decompressed size of more than 8192 bytes.

Comment (alexander-beedie) [reply py-polars/tests/unit/io/test_csv.py:1744]: > This issue only triggers when reading a compressed file with a decompressed size of more than 8192 bytes.

There's a trick to rapidly generate large test frames, which you can use to create sufficient CSV data for the test, like so:
```python
df = pl.DataFrame({
  "idx": range(2_500_000),
  "dt" : date(2025,12,31), 
  "txt": "hello world",
})
csv_data = df.write_csv()
```
This only takes ~0.175 secs on my laptop and creates uncompressed data of ~60MB. You could then gz-compress it to run through the validating code 👌

Comment (Wainberg) [reply py-polars/tests/unit/io/test_csv.py]: I like that approach! Just implemented it in the test.

**Comment by Wainberg** [comment]
I think this one's ready to merge now too :)

**Review by ritchie46 — APPROVED** [review]

**Pull request merged**